crates/scarb-api/src/lib.rs

```diff
@@ -75,64 +78,121 @@ fn artifacts_for_package(path: &Utf8Path) -> Result<StarknetArtifacts> {
     Ok(starknet_artifacts)
 }
 
+#[derive(PartialEq, Debug)]
+struct ContractArtifactData {
+    path: Utf8PathBuf,
+    test_type: Option<String>,
+}
+
+fn build_target_path(target_dir: &Utf8Path, file_name: &str) -> Option<Utf8PathBuf> {
+    let path = target_dir.join(file_name);
+
+    if path.exists() {
+        Some(path)
+    } else {
+        None
+    }
+}
+
+fn get_starknet_artifacts_paths_from_test_targets(
+    target_dir: &Utf8Path,
+    test_targets: &HashMap<String, &TargetMetadata>,
+) -> Vec<ContractArtifactData> {
+    let starknet_artifacts_file_name =
+        |name: &str, metadata: &TargetMetadata| -> Option<ContractArtifactData> {
+            let path = format!("{name}.test.starknet_artifacts.json");
+            let path = build_target_path(target_dir, &path);
+            let test_type = metadata
+                .params
+                // TODO get rid of hardcoding
+                .get("test-type")
+                .and_then(|value| value.as_str())
+                .unwrap();
```

Comment:
I don't think this unwrap is safe, is it? Is it guaranteed that the `test-type` will be known for manually defined test targets?

Reply:
I think I'll just default to `None` since I'd have to hardcode a string here.